File: src/main/java/com/sgorokh/TimeSaver/controllers/responses/ClientDetailsResponse.java
package com.sgorokh.TimeSaver.controllers.responses;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Builder
@Data
@AllArgsConstructor
public class ClientDetailsResponse {

    private Long id;

    private String name;

    private String phone;

    private String email;

    private List<SessionDetails> sessions;

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    @Builder
    public static class SessionDetails {

        private Long id;

        private String name;

        private Long imageId;
    }
}

File: frontend/src/app/shared/responses/client-details.response.ts

export class ClientDetailsResponse {

  id: number;
  name: string;
  phone: string;
  email: string;
  sessions: SessionDetail[];

  constructor(id:number, name: string, email: string, phone: string, sessions: SessionDetail[]) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.phone = phone;
    this.sessions = sessions;
  }

}

export class SessionDetail {

  id: number;
  name: string;
  imageId: number;

  constructor(id: number, name: string, imageId: number) {
    this.id = id;
    this.name = name;
    this.imageId = imageId;
  }
}

export class SessionDetailWithImage {

  id: number;
  name: string;
  image: string;

  constructor(id: number, name: string, image: string) {
    this.id = id;
    this.name = name;
    this.image = image;
  }
}
